flagscale/runner/runner_train.py

```diff
@@ -382,38 +416,31 @@ def run(self, with_test=False, dryrun=False, monitor=False, interval=10):
             nnodes = get_nnodes(nnodes_from_hostfile, nnodes_from_args)
             available_ip = list(self.resources.keys())[0]
             available_port = get_free_port()
-            for node_rank, (host, resource_info) in enumerate(self.resources.items()):
-                if node_rank >= nnodes:
-                    break
-                self.cur_envs = add_decive_extra_config(
-                    self.user_envs, resource_info["type"]
-                )
-                visible_devices = self.cur_envs.get("CUDA_VISIBLE_DEVICES", None)
-                if visible_devices is not None and isinstance(visible_devices, str):
-                    visible_devices = visible_devices.split(",")
-                    num_visible_devices = len(visible_devices)
-                nproc_from_hostfile = resource_info["slots"]
-                nproc_from_args = runner_config.get("nproc_per_node", None)
-                nproc_per_node = get_nproc_per_node(
-                    nproc_from_hostfile, nproc_from_args, num_visible_devices
-                )
-                master_addr = runner_config.get("master_addr", available_ip)
-                master_port = runner_config.get("master_port", available_port)
-                self._run_each(
-                    host,
-                    master_addr,
-                    master_port,
-                    nnodes,
-                    node_rank,
-                    nproc_per_node,
-                    device_type=resource_info["type"],
-                    with_test=with_test,
-                    dryrun=dryrun,
-                )
+            with multiprocessing.Pool(processes=nnodes) as pool:
```

Comment:
Ensure the pool processes match the number of available processors. It's best to automatically detect and use that number.

Reply:
DONE.